src/CG/CG_elem.cpp

```diff
@@ -2113,6 +2113,829 @@ bool intersectSegmentPolygon( array3D const &P0, array3D const &P1, std::size_t
     return false;
 }
 
+/*!
+ * Check if a pixel is crossing a given plane and fully calculate the intersection (a segment) with 
+ * edges, eventually. A pixel is defined as a planar rectangular, vertical to a cartesian axis.
+ * The polygon greated from the intersection belongs to the subspace opposite of the
+ * one pointed by the plane's normal
+ *
+ * Given the normal cartesian direction dir, the pixel coordinates
+ * are represened on its 2D plane as (x, y) where x = (dim + 1) % 3 and y = (dim + 2) % 3 
+ *
+ * Local indeces for edges are defined as follows:
+ * edge 0: y = y min
+ * edge 1: x = x max
+ * edge 2: y = y max
+ * edge 3: x = x min
+ *
+ * Co-planarity of input pixel and plane will produce no intersection.
+ * No intersection is considered if the plane passes from a pixel's vertex,
+ * and no other intersection exist.
+ *
+ * \param[in] Pp point belonging to the plane
+ * \param[in] nP unit normal to plane
+ * \param[in] A0 min point of pixel on the 2D space defined by the pixel direction dir
+ * \param[in] A1 max point of pixel on the 2D space defined by the pixel direction dir
+ * \param[in] dir is an integer from 1 to 3 and indicates the cartesian direction normal
+ * to the pixel. Its sign affects the polynomial's orientation. The above used dim variable
+ * is computed as dim = |dir| - 1
+ * \param[in] coor indicates the cartesian coordinate of all the pixel points
+ * \param[out] intersection If intersects, return the cutting segment
+ * \param[out] edges_of_intersection If intersects, return the pixel edge local indices, 
+ * of the intersected edges. If no intersection occurs return a -1 label. 
+ * \param[in] distanceTolerance if distance among features exceed this value they are considered as not intersecting
+ * \return if intersects
+ */
+bool intersectPlanePixel(array3D const &Pp, array3D const &nP, std::array<double, 2> const &A0,
+                         std::array<double, 2> const &A1, int dir, double coor, std::array<array3D, 2> &intersection,
+                         std::array<int, 2> &edges_of_intersection, const double distanceTolerance)
```

Comment:
Is this function much faster than intersectPlanePolygon? In the CG module this will be the only function that deals with pixels. We can just drop it and call intersectPlanePolygon for pixels too.

Reply:
It's 1.3 times faster. For me, this number is enough to keep the handling of pixels separately. This function is going to be called thousands of times during the cut-cell mesh generation. Usually, this is one of the slowest parts during this process and must be as fast as possible. If you prefer, I can add intersections of pixels with other geometrical elements, so the library will be more complete.